sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options related to logging.
+/// </summary>
+public class LoggingOptions
+{
+    private bool _frozen;
+
+    private const int DefaultLoggedContentSizeLimit = 4 * 1024;
+    private const bool DefaultIsLoggingContentEnabled = false;
+    private const bool DefaultCombineLogs = true;
+
+    private static readonly string[] s_defaultAllowedHeaderNames =
+        new[] {
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" };
+    private static readonly string[] s_defaultAllowedQueryParameters = new[] { "api-version" };
+
+    private int _loggedContentSizeLimit = DefaultLoggedContentSizeLimit;
+    private bool _isLoggingContentEnabled = DefaultIsLoggingContentEnabled;
+    private IList<string> _allowedHeaderNames = new List<string>(s_defaultAllowedHeaderNames);
+    private IList<string> _allowedQueryParameters = new List<string>(s_defaultAllowedQueryParameters);
+    private ILoggerFactory _loggerFactory = NullLoggerFactory.Instance;
+
+    /// <summary>
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
```

Comment:
TODO - should add better docs that explain that if an ILogger is passed in, nothing will be written to event source